### PR TITLE
Adds functions to create and delete Atlassian users from Demand Management ServiceDesk queues

### PR DESCRIPTION
**Pull request by DannyGB**

### CODE REVIEW

File: src/SFA.DAS.OnBoardingTool.Web/wwwroot/css/main.css [hunk 1]
```diff
@@ -0,0 +1 @@
...ize-16{font-size:14pt !important;line-height:1.2 !important}}.govuk-\!-font-size-14{font-size:12px !important;font-size:.75rem !important;line-height:1.25 !important}@media (min-width: 40.0625em){.govuk-\!-font-size-14{font-size:14px !important;font-size:.875rem !important;line-height:1.42857 !important}}@media print{.govuk-\!-font-size-14{font-size:12pt !important;line-height:1.2 !important}}.govuk-\!-font-weight-regular{font-weight:400 !important}.govuk-\!-font-weight-bold{font-weight:700 !important}.govuk-\!-width-full{width:100% !important}.govuk-\!-width-three-quarters{width:100% !important}@media (min-width: 40.0625em){.govuk-\!-width-three-quarters{width:75% !important}}.govuk-\!-width-two-thirds{width:100% !important}@media (min-width: 40.0625em){.govuk-\!-width-two-thirds{width:66.66% !important}}.govuk-\!-width-one-half{width:100% !important}@media (min-width: 40.0625em){.govuk-\!-width-one-half{width:50% !important}}.govuk-\!-width-one-third{width:100% !important}@media (min-width: 40.0625em){.govuk-\!-width-one-third{width:33.33% !important}}.govuk-\!-width-one-quarter{width:100% !important}@media (min-width: 40.0625em){.govuk-\!-width-one-quarter{width:25% !important}}
```
Comment (CodeWolf86) [inline src/SFA.DAS.OnBoardingTool.Web/wwwroot/css/main.css]: Do you not want to be referencing the latest front end package?

File: src/SFA.DAS.OnBoardingTool.Web/Pages/Success.cshtml.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace MyApp.Namespace
```
Comment (CodeWolf86) [inline src/SFA.DAS.OnBoardingTool.Web/Pages/Success.cshtml.cs]: Should the namespace be something different to MyApp ?

File: src/SFA.DAS.OnBoardingTool.Web/Pages/Success.cshtml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+﻿@page
+@model MyApp.Namespace.SuccessModel
```
Comment (CodeWolf86) [inline src/SFA.DAS.OnBoardingTool.Web/Pages/Success.cshtml]: Should the namespace be something different to MyApp ?

File: src/SFA.DAS.OnBoardingTool.Infrastructure/Extensions/UriBuilderExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            if (queryParams?.Count == 0)
+            {
+                uriBuilder.Query = String.Empty;
+            }
+            else
+            {
+                var sb = new StringBuilder();
+                var first = true;
+                foreach(var keyVal in queryParams)
+                {
```
Comment (CodeWolf86) [inline src/SFA.DAS.OnBoardingTool.Infrastructure/Extensions/UriBuilderExtensions.cs:22]: Could this be simplified?
```
if(first)
{
   sb.Append("?");
} 
else
{
 sb.Append('&')
}

File: src/SFA.DAS.OnBoardingTool.Application/Atlassian/Services/AtlassianService.cs [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+
+            _log.LogInformation($"Finished processing create requests");
+        }
+
+        public async Task ProcessRevokeUserRequests()
+        {
+            _log.LogInformation(_config.ToString());
+            _log.LogInformation($"Processing revoke user requests");
+
+            var requests = await GetUserRequests();
```
Comment (CodeWolf86) [inline src/SFA.DAS.OnBoardingTool.Application/Atlassian/Services/AtlassianService.cs:84]: I'm guessing the Api doesn't have any query parameters to only find the requests that have a custom field value of "revoke" or "grant" to remove the Where restriction on the loop?

File: src/SFA.DAS.OnBoardingTool.Application/Atlassian/Services/AtlassianService.cs [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+            {
+                _log.LogInformation($"No revoke user requests to process");
+                return;
+            }
+
+            var tasks = new List<Task>();
+
+            foreach (var request in requests.Values
+                .AsEnumerable()
+                .Where(r => r.Fields.Customfield_12105.Value.Equals("revoke", StringComparison.InvariantCultureIgnoreCase)))
```
Comment (CodeWolf86) [inline src/SFA.DAS.OnBoardingTool.Application/Atlassian/Services/AtlassianService.cs:96]: Your MMV, I prefer the coding style of declaring the list up front, 

```
var revokeRequests = requests.ValuesAsEnumerable()Where(r => r.Fields.Customfield_12105.Value.Equals("revoke", StringComparison.InvariantCultureIgnoreCase))`
                
                foreach(var request in revokeRequests)
```
                
As I think you get a neater declaration, and descriptive variable of the held list. This does give a negligible performance benefit but it's hardly worth mentioning as a for or against as my main item is the name and command separation

**Review by CodeWolf86 — APPROVED** [review]

File: src/SFA.DAS.OnBoardingTool.Application/Atlassian/Services/AtlassianService.cs (hunk 5 above)
Comment (DannyGB) [reply src/SFA.DAS.OnBoardingTool.Application/Atlassian/Services/AtlassianService.cs:84]: Correct there's no way (currently) to filter that at the API level

File: azure-pipelines.yml [hunk 7]
```diff
@@ -25,12 +22,32 @@ resources:
...
+  - repository: das-platform-automation
+    type: github
+    name: SkillsFundingAgency/das-platform-automation
+    ref: refs/tags/4.4.0
+    endpoint: GitHub (SFA)
+  pipelines:
+  - pipeline: das-employer-config
+    project: Digital Apprenticeship Service
+    source: das-employer-config
+    branch: das-onboarding-initial-config
```
Comment (adam230594) [inline azure-pipelines.yml]: is this branch in master yet?

Comment (cpwevans) [reply azure-pipelines.yml]: I have spoken to Dan and he says when this PR is approved the config will be pushed to Master

**Review by adam230594 — APPROVED** [review]

**Pull request merged**